doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional.
+However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
```

Comment:
Depends on what you mean by "check". AFAIK there is no single check for Hilbert space anywhere (e.g. exponent 2 in `DiscreteLp`), so things run smoothly even for non-Hilbert cases. They may just not be mathematically valid any longer. 

What we can do here is weaken the statement and say that what one needs is a dual pairing of real Banach spaces, and that the available proximals have been implemented with L^2 spaces in mind. Validity for other spaces is not guaranteed.